llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/InstrInterval.h

```diff
@@ -80,6 +80,17 @@ class InstrInterval {
     assert((FromI == ToI || FromI->comesBefore(ToI)) &&
            "FromI should come before TopI!");
   }
+  InstrInterval(ArrayRef<Instruction *> Instrs) {
+    assert(!Instrs.empty() && "Expected non-empty Instrs!");
+    FromI = Instrs[0];
+    ToI = Instrs[0];
+    for (auto *I : drop_begin(Instrs)) {
+      if (I->comesBefore(FromI))
+        FromI = I;
+      if (ToI->comesBefore(I))
```

Comment:
else if here?  Either FromI moves or ToI moves but not both.

Reply:
I think you are right. Initially they both have the same value and from that point on FromI will always be before ToI, so if I comes before FromI it will never be after ToI